### PR TITLE
Activate Pylint Rules

### PR DESCRIPTION
**Pull request by zariiii9003**
Can you check these?

```text
  style: commands[1]> ruff check ./src
  src/asammdf/gui/widgets/plot.py:136:8: PLR0124 Name compared with itself, consider replacing `b != b`
  src/asammdf/gui/widgets/plot.py:144:8: PLR0124 Name compared with itself, consider replacing `b != b`
  src/asammdf/gui/widgets/plot.py:4460:28: PLR0124 Name compared with itself, consider replacing `min_ != min_`
  src/asammdf/gui/widgets/plot.py:4462:28: PLR0124 Name compared with itself, consider replacing `max_ != max_`
  src/asammdf/gui/widgets/plot.py:4503:28: PLR0124 Name compared with itself, consider replacing `min_ != min_`
  src/asammdf/gui/widgets/plot.py:4505:28: PLR0124 Name compared with itself, consider replacing `max_ != max_`
  src/asammdf/gui/widgets/tabular_base.py:1382:16: PLR0124 Name compared with itself, consider replacing `target != target`
  Found 7 errors.
```

https://github.com/danielhrisca/asammdf/blob/43852ffec3707e835a5bf401851e65561620c944/src/asammdf/gui/widgets/plot.py#L136
https://github.com/danielhrisca/asammdf/blob/43852ffec3707e835a5bf401851e65561620c944/src/asammdf/gui/widgets/plot.py#L144
https://github.com/danielhrisca/asammdf/blob/43852ffec3707e835a5bf401851e65561620c944/src/asammdf/gui/widgets/plot.py#L4460
https://github.com/danielhrisca/asammdf/blob/43852ffec3707e835a5bf401851e65561620c944/src/asammdf/gui/widgets/plot.py#L4462
https://github.com/danielhrisca/asammdf/blob/43852ffec3707e835a5bf401851e65561620c944/src/asammdf/gui/widgets/plot.py#L4503
https://github.com/danielhrisca/asammdf/blob/43852ffec3707e835a5bf401851e65561620c944/src/asammdf/gui/widgets/plot.py#L4505
https://github.com/danielhrisca/asammdf/blob/43852ffec3707e835a5bf401851e65561620c944/src/asammdf/gui/widgets/tabular_base.py#L1382

### CODE REVIEW

**Comment by danielhrisca** [comment]
Those comparison are to check is the value is not a number NaN

**Comment by zariiii9003** [comment]
> Those comparison are to check is the value is not a number NaN

Ah, i didn't know that. Would you mind if i replace it with np.isnan for readability?

**Comment by danielhrisca** [comment]
My concern is the execution speed in this case

image

File: src/asammdf/blocks/mdf_v4.py [hunk 1]
```diff
@@ -5693,15 +5693,13 @@ def extend(self, index: int, signals: list[tuple[NDArray[Any], NDArray[Any] | No
...
-                            data.append(elem)
+                            data.extend((UINT32_p(size), elem))
                             off += size + 4
                     else:
                         for elem in signal:
                             offsets.append(off)
                             size = len(elem)
-                            data.append(UINT32_p(size))
-                            data.append(elem)
+                            data.extend((UINT32_p(size), elem))
```
Comment (danielhrisca) [inline src/asammdf/blocks/mdf_v4.py:5702]: Is this automatically done by ruff?

File: src/asammdf/blocks/mdf_v4.py [hunk 2]
```diff
@@ -6593,7 +6590,7 @@ def get(
             if name is None:
                 name = channel.name
 
-            unit = conversion and conversion.unit or channel.unit
+            unit = conversion.unit if conversion else channel.unit
```
Comment (danielhrisca) [inline src/asammdf/blocks/mdf_v4.py:6593]: This is not right: if the conversion unit is empty then we need to check the channel unit as well

File: src/asammdf/gui/dialogs/advanced_search.py [hunk 3]
```diff
@@ -174,7 +174,7 @@ def search_text_changed(self):
                                     (group_index, channel_index): {
                                         "names": [ch.name],
                                         "comment": extract_xml_comment(ch.comment).strip(),
-                                        "unit": ch.conversion and ch.conversion.unit or ch.unit,
+                                        "unit": ch.conversion.unit if ch.conversion else ch.unit,
```
Comment (danielhrisca) [inline src/asammdf/gui/dialogs/advanced_search.py:177]: same as above, also applies to the following commit diffs

File: src/asammdf/gui/widgets/tabular_base.py [hunk 4]
```diff
@@ -1379,30 +1379,20 @@ def apply_filters(self, event=None):
             column_name = pandas_query_compatible(column_name)
             op = filter.op.currentText()
 
-            if target != target:
-                # here we have NaN
-                nan = np.nan
```
Comment (danielhrisca) [inline src/asammdf/gui/widgets/tabular_base.py:1384]: The pandas query check the local scope, so this has to remain otherwise `@nan` will fail

**Review by danielhrisca — CHANGES_REQUESTED** [review]
please address the open topics

File: src/asammdf/blocks/mdf_v4.py (hunk 1 above)
Comment (zariiii9003) [reply src/asammdf/blocks/mdf_v4.py:5702]: Not really, this actually not part of the pylint rules, instead i ran some of the preview rules: https://docs.astral.sh/ruff/rules/repeated-append/
Even then this is only done automatically if you call ruff with `--unsafe-fixes`. 
Should i revert these?

Comment (danielhrisca) [reply src/asammdf/blocks/mdf_v4.py:5702]: It's ok I guess, it was more a curiosity

File: src/asammdf/blocks/mdf_v4.py (hunk 2 above)
Comment (zariiii9003) [reply src/asammdf/blocks/mdf_v4.py:6593]: True. I just found out, that the preview rule, which flagged this, will be [removed soon](https://github.com/astral-sh/ruff/issues/9007)

Comment (zariiii9003) [reply src/asammdf/blocks/mdf_v4.py]: Fixed

File: src/asammdf/gui/dialogs/advanced_search.py (hunk 3 above)
Comment (zariiii9003) [reply src/asammdf/gui/dialogs/advanced_search.py]: Fixed

File: src/asammdf/gui/widgets/tabular_base.py (hunk 4 above)
Comment (zariiii9003) [reply src/asammdf/gui/widgets/tabular_base.py:1384]: Fixed

**Review by danielhrisca — APPROVED** [review]

**Pull request merged**